aemcmc/negative_binomial_horseshoe_gibbs.py

```diff
@@ -0,0 +1,199 @@
+import aesara
+import aesara.tensor as at
+from aesara.tensor.var import TensorVariable
+
+from aemcmc.dists import polyagamma
+
+
+# this technique is very unstable for this samplers because while sampling
+# lambda2 and tau2 can become large quickly and this means `A` gets less and less
+# positive-definite and eventually the cholesky descomposition fails and
+# returns NaNs and this raises an exception which halts the sampling.
+def _update_beta2(rng, omega, lambda2tau2, X, z):
```

Comment:
We should consider how we want to organize this repository, because these two functions would serve best as generic multivariate normal samplers.  For instance, we could have them in a `utils.py` module, assuming that they'll be used by more than one other Gibbs sampler eventually.

Reply:
Is there a reason you'd choose a `utils.py` rather than the `dists.py` module?

Reply:
It definitely doesn't need to be `utils`.